bin/check-http.rb

```diff
@@ -28,6 +28,12 @@
 #   Use a proxy to check a URL
 #   check-http.rb -u https://www.google.com --proxy-url http://my.proxy.com:3128
 #
+#   Use a proxy with username and password to check a URL - unsafe! Shows credentials in 'ps'
+#   check-http.rb -u https://www.google.com --proxy-url http://a_user:a_pass@my.proxy.com:3128
+#
+#   Use a proxy with username and password to check a URL - safe
+#   PROXY_USER=a_user PROXY_PASS=a_pass check-http.rb -u https://www.google.com --proxy-url http://my.proxy.com:3128
```

Comment:
This would still be exposed if specified in the same command, you could try using env vars passed in when the sensu process starts up. What I was really referring to is a feature called token substitution. You can read about it here: https://sensuapp.org/docs/latest/reference/checks.html#check-token-substitution

Basically you define the value in the client and then on the cli you use `:::var.name:::` to ensure that it is protected from the process table. you can further ensure that it never leaves the client by using the `redact` attribute: https://sensuapp.org/docs/1.1/reference/clients.html#client-attributes, one thing to note is that you want to specify any values to redact so make sure to include the default ones unless you specifically don't want those.

Reply:
Aha! I did not know what token substitution was - very good to know, thanks for the explanation.
I have removed the extra code and modified the comment! Thanks!